sashimi/state.py

```diff
@@ -646,6 +658,15 @@ def calculate_pulse_times(self):
             / (self.volume_setting.frequency * self.volume_setting.n_planes)
         )
 
+    def set_trigger_mode(self, mode: bool):
+        if mode:
+            self.external_comm.is_triggered_event.set()
+        else:
+            self.external_comm.is_triggered_event.clear()
+
+    def send_manual_duration(self):
```

Comment:
The flow is as follows. If the widget is checked in the GUI then everything proceeds like before. When unchecked, two things happen:

1. The `State` sets a event in the `ExternalComm` process indicating it to not put the duration on the queue (which happens only if the experiment is set to start)
2. Duration widget is set to `enabled` &mdash; the change in `lightparam.Param` is connected to the function you point to but it does not really send anything while the box is checked because the widget is disabled.

Does this answer your question?